### PR TITLE
Implement Algolia search

### PR DESCRIPTION
**Pull request by anik120**
This PR introduces search powered by [algolia](https://algolia.com)

### CODE REVIEW

File: assets/js/app.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Set Algolia options
+const options = {
+    appId: "UI9O9HIU4I",
+    apiKey: "6a380ea9bf30cec543bb36e2b3bdee01",
```
Comment (exdx) [inline assets/js/app.js:4]: should this be in git?

Comment (anik120) [reply assets/js/app.js:4]: @exdx this is the search API Key, algolia has a ADMIN API key that allows you to load indexes that they recommend keeping a secret. 

The cavate here is that anyone with the search API key can use it to hit up our API, and algolia has a free tier that allows for 1000 search for free. So we have to make sure that those 1000 searches are from visitors to our sites and not anyone playing around with the API, but I haven't found a elegant solution for this

File: layouts/partials/navbar.html [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
+      {{ end }}
+      {{ if  .Site.Params.versions }}
+      <li class="of-nav-main__item">
+      {{ partial "navbar-version-selector.html" . }}
+      </li>
+      {{ end }}
+    </ul>
+  </nav>
+  <div id="search-box">
+    <!-- SearchBox widget will appear here -->
```
Comment (matthewcarleton) [inline layouts/partials/navbar.html:23]: try dropping this in here
`{{ partial "search-input.html" . }}`

Comment (anik120) [reply layouts/partials/navbar.html:23]: @matthewcarleton do you mean like thishttps://github.com/operator-framework/olm-docs/blob/d352a7f7771ae32b6fa1308143e85e2cc0e72335/layouts/partials/navbar.html#L24? 

Isn't that just calling `search-input.html`? 
I was under the impression that this algolia search box will be replacing the `search-input.html`

**Comment by ecordell** [comment]
Should the search work on the preview site?

**Comment by matthewcarleton** [comment]
> Should the search work on the preview site?

are you saying the search could not be showing up b/c it's just a preview?

**Comment by anik120** [comment]
@matthewcarleton @ecordell I'll merge this if you guys don't have any objection

**Review by hasbro17 — APPROVED** [review]
LGTM

**Pull request merged**